gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCalls.java

```diff
@@ -48,12 +50,19 @@ public static <RequestT, ResponseT> HttpJsonClientCall<RequestT, ResponseT> newC
       ApiMethodDescriptor<RequestT, ResponseT> methodDescriptor, ApiCallContext context) {
 
     HttpJsonCallContext httpJsonContext = HttpJsonCallContext.createDefault().nullToSelf(context);
+    HttpJsonCallOptions callOptions = httpJsonContext.getCallOptions();
+
+    Map<String, String> headerMap = callOptions.getRequestHeaderMap();
+    for (Map.Entry<String, List<String>> extraHeaderEntrySet :
+        httpJsonContext.getExtraHeaders().entrySet()) {
+      headerMap.put(extraHeaderEntrySet.getKey(), extraHeaderEntrySet.getValue().get(0));
```

Comment:
I believe the Headers class (used when merging the headers in `withExtraHeaders`) has a check for non-null: https://github.com/googleapis/sdk-platform-java/blob/ef4a7ff91b259f0b650cc06bd664eadca7bd17c9/gax-java/gax/src/main/java/com/google/api/gax/rpc/internal/Headers.java#L50-L52

I can add a check so that there is one element inside.

Reply:
In addition, I'm not sure why the value is of type `List<String>`, my guess is that we did it on grpc side and just replicated the same pattern, and the only reason we did it on grpc side is that it's what `grpc-java` needed, maybe they concatenate the values like how we did it in `RequestUrlParamsEncoder`.
Maybe not the scope of this PR, but if we want to start support this `extraHeaders`, it's better to support it properly, either just always use the first element, change the value to `String` or concatenate the values.

Reply:
> it's better to support it properly, either just always use the first element, change the value to String or concatenate the values.

Yeah, agreed. I don't fully understand the `List<String>` pattern for headers either. I wanted to update to just a (K,V) of <String, String> but that is a breaking change on a `@BetaApi`. I'll do some more digging into this and see if I can find a reason for this. Depending on the context, do you think this would be something that we *could* consider modifying?

Reply:
This could be a follow up to this PR, I'm OK with always getting the first element for now in this PR. We can have a separate issue just for supporting `extraHeaders`.

Reply:
Doing a little digging and I believe this is probably the intended behavior for grpc (https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#constructing-metadata) - One header key can support multiple values. Seems like we copied the behavior for httpjson and that's why we have a `Map<String, List<String>>`.

I can create a separate issue for looking into `extraHeaders` and seeing if copying grpc's behavior is needed and/or how we can better support it if it is.